src/intersection-observer.js

```diff
@@ -208,6 +209,10 @@ export class IntersectionObserver extends Observable {
    * @private
    */
   flush_() {
+    if (!this.isElementInDoc_()) {
```

Comment:
> I assume it's not super expensive, but would still need to access win.document, so yes I would prefer not to have it.

Anything that touches the DOM is expensive.

> I think what you are suggesting is call destroy when that element is removed. So how can I know when and where that's happening,

We'd have to audit anywhere that's using an `IntersectionObserver`, but that's only `<amp-ad>`, `<amp-a4a>`, and `<amp-iframe>`. This is a much better choice, though.